File: ayrfu-frontend/src/api/services/client.service.js
//src/api/services/client.service.js
import axios from '../axios';

const clientService = {
  getAllClients: async () => {
    return axios.get('/clients');
  },

  getClientById: async (id) => {
    return axios.get(`/clients/${id}`);
  },

  getClientByEmail: async (email) => {
    return axios.get(`/clients/email/${email}`);
  },

  getCurrentClient: async () => {
    return axios.get('/users/profile/client');
  },

  createClient: async (clientData) => {
    return axios.post('/clients', clientData);
  },

  updateClient: async (id, clientData) => {
    return axios.put(`/clients/${id}`, clientData);
  },

  deleteClient: async (id) => {
    return axios.delete(`/clients/${id}`);
  }
};

export default clientService;

File: ayrfu-backend/src/main/java/com/ayrfu/ayrfu/service/ClientService.java
package com.ayrfu.ayrfu.service;


import com.ayrfu.ayrfu.dto.request.ClientRequest;
import com.ayrfu.ayrfu.dto.response.ClientResponse;

import java.util.List;

public interface ClientService {

    ClientResponse createClient(ClientRequest clientRequest);

    ClientResponse getClientById(Long id);

    ClientResponse getClientByEmail(String email);

    List<ClientResponse> getAllClients();

    ClientResponse updateClient(Long id, ClientRequest clientRequest);
    boolean isOwnProfile(Long clientId, Long userId);
    void deleteClient(Long id);
}
